dev/com.ibm.ws.microprofile.metrics.3.0/src/com/ibm/ws/microprofile/metrics30/impl/SimpleTimer30Impl.java

```diff
@@ -0,0 +1,204 @@
+         * not reset the start time. Multiple calls result in multiple updates.
+         *
+         * @return the elapsed time in nanoseconds
+         */
+        @Override
+        public long stop() {
+            final long elapsed = clock.getTick() - startTime;
+            simpleTimer.update(Duration.ofNanos(elapsed));
+            return elapsed;
+        }
+
+        /** Equivalent to calling {@link #stop()}. */
+        @Override
+        public void close() {
+            stop();
+        }
+    }
+
+    private final Counter count;
+    private Duration elapsedTime;
+    private final Clock clock;
+
+    public SimpleTimer30Impl() {
+        this(Clock.defaultClock());
+    }
+
+    public SimpleTimer30Impl(Clock clock) {
+        this.clock = clock;
+        this.count = new CounterImpl();
+        this.elapsedTime = Duration.ZERO;
+
+        max_previousMinute = new AtomicLong(0);
+        min_previousMinute = new AtomicLong(0);
+        max_thisMinute = new AtomicLong(0);
+        min_thisMinute = new AtomicLong(0);
+        thisMinute = new AtomicLong(getCurrentMinuteFromSystem());
+    }
+
+    /**
+     * Adds a recorded duration.
+     *
+     * @param duration the length of the duration
+     * @param unit     the scale unit of {@code duration}
+     */
+    @Override
+    public void update(Duration duration) {
+        maybeStartNewMinute();
+        Long duration_nanos = duration.toNanos();
+        if (duration_nanos >= 0) {
+            synchronized (this) {
+                count.inc();
+                elapsedTime = elapsedTime.plus(duration);
+
+                if (duration_nanos > max_thisMinute.get())
+                    max_thisMinute.set(duration_nanos);
+
+                if (duration_nanos < min_thisMinute.get() || min_thisMinute.get() == 0L)
+                    min_thisMinute.set(duration_nanos);
+            }
+        }
+    }
+
+    @Override
+    public <T> T time(Callable<T> event) throws Exception {
+        maybeStartNewMinute();
+        final long startTime = clock.getTick();
+        try {
+            return event.call();
+        } finally {
+            update(Duration.ofNanos(clock.getTick() - startTime));
+        }
+    }
+
+    @Override
+    public void time(Runnable event) {
+        maybeStartNewMinute();
+        final long startTime = clock.getTick();
+        try {
+            event.run();
+        } finally {
+            update(Duration.ofNanos(clock.getTick() - startTime));
+
+        }
+    }
+
+    @Override
+    public Context time() {
+        maybeStartNewMinute();
+        return new Context(this, clock);
+    }
+
+    @Override
+    public long getCount() {
+        maybeStartNewMinute();
+        return count.getCount();
+    }
+
+    @Override
+    public Duration getElapsedTime() {
+        maybeStartNewMinute();
+        return elapsedTime;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Duration getMaxTimeDuration() {
```

Comment:
The implementation for calculating/retrieving the previous max high and low is similar to the implementation from ConcurrentGauge.